deployments/mainnet/Dai.json

```diff
@@ -0,0 +1,225 @@
+{
+    "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
```

Comment:
Ok, I think that storing all this files with the ABI on the github is an overkill. More ledgitimate way will be to have one JSON file, that will have all contracts in the simple format: `name`:`address`. That's it. Full jsons should be stored only locally, so I think preferably, this folder should be added to .gitignore

Reply:
This PR is an attempt to reduce the amount of overkill on github and make our future PRs cleaner. Why I think this makes sense can be read in the issue. I'll try to respond to your two concerns: 1. overkill and 2. local storage of deployments. 

First, the number of JSON files (and code in general) that we are storing on github is being reduced by roughly 1 million lines through this PR. When this PR is merged we will only ever store three files for the dApp development on GitHub (? the `exports` directory). And these three files are optimized to be lightweight per the [hardhat export functionality](https://hardhat.org/plugins/hardhat-deploy.html#exporting-deployments) (so for instance no unnecessary bytecode is checked into github, just ABI and address) and - as result of Doug's feedback - per an additional custom logic which takes care that for two contracts that share the same ABI, we store the ABI only once and let both contracts point to that shared ABI. The `/build` directory alone has a size of 30mb and we do not need to check it in anymore. Compare that with the combined sizes of the `deployments` and `exports` folders of 1.6mb.

Second, adding `deployments` to the .gitignore would be possible, but I really do not see the point why we should do this. `deployments` has a size of 1.5mb and when checking it in everyone can easily interact with the same contracts. If I deploy some contracts to rinkeby and you pull the code, you have the same deployed contracts available. You can call the functions of the same deployed contract instances as I do via the available hardhat tasks. These hardhat tasks fully depend on the deployments folder. So at least for the mainnet deployments we would need to manually insert the deployed contracts to be able to interact w/ them via hardhat tasks. fwiw [sushiswap](https://github.com/sushiswap/sushiswap) takes the same approach.

Reply:
I think one could argue that putting WETH and DAI into the `deployments` folder is too much, since, to my knowledge, we never really interact with these contracts in our project. Afaiu, their addresses/abis are needed for dApp development. That's why included them in `deployments`, so that `hardhat export` could take care that their information are exported. We could remove such "external" contracts from deployments and inject their information into the export files manually through the `exportAbis.js` script. Wdyt?

Reply:
We aren't deploying those tokens, so why would they be in "deployment"?

But the dApps do need those contract addresses.

Reply:
> We aren't deploying those tokens, so why would they be in "deployment"?
> 
> But the dApps do need those contract addresses.

I agree. I will remove them from the folder and proceed as described in my [previous comment](https://github.com/PrimeDAO/contracts-v2/pull/109#discussion_r694601999).

Reply:
Just to keep in mind:  The dApps do need the ERC20 artifacts given "WETH" or "DAI".

Reply:
Ok, so as I understand we have hardhat tasks that are dependent on this folder, then it'll take to much to change I suppose, let it be...